pkgs/development/tools/misc/luarocks/default.nix

```diff
@@ -66,7 +66,7 @@ stdenv.mkDerivation (finalAttrs: {
   postInstall = ''
     sed -e "1s@.*@#! ${lua}/bin/lua$LUA_SUFFIX@" -i "$out"/bin/*
     substituteInPlace $out/etc/luarocks/* \
-     --replace '${lua.luaOnBuild}' '${lua}'
```

Comment:
@Artturin just wanted to confirm with you this was the correct choice and wont explode on crosscompilation

Reply:
This seems to have caused a regression:

```bash
nix build nixpkgs#luajit_openresty.pkgs.luarocks_bootstrap
```
> substituteStream() in derivation luarocks_bootstrap-3.11.1: ERROR: pattern /nix/store/4qkdd4slrr3i8rsk0zj4kk0rmyqmbli4-luajit-2.1-20220915 doesn't match anything in file '/nix/store/g3w19f2xclqqh6yn9rf1vcnxi5icbjgx-luarocks_bootstrap-3.11.1/etc/luarocks/config-5.1.lua'
> error: builder for '/nix/store/v700wnxlzmvxsxizc86bc4q1bljcyisk-luarocks_bootstrap-3.11.1.drv' failed with exit code 1

Reply:
Hmm that file is empty, I was wondering why it was added and found that it was added by me :D https://github.com/NixOS/nixpkgs/commit/7b24b57c14b954ff15a0b49b8a0404a222e3f258

`pkgsCross.aarch64-multiplatform.luaPackages.luarocks_bootstrap` contains 

```
-- LuaRocks configuration

rocks_trees = {
   { name = "user", root = home .. "/.luarocks" };
   { name = "system", root = "/nix/store/zsqak897x5y75i3jh9brih1pd3silpih-luarocks_bootstrap-aarch64-unknown-linux-gnu-3.11.1" };
}
variables = {
   LUA_DIR = "/nix/store/29p4fvkmbr6hy91304dcsragir7k7bmm-lua-aarch64-unknown-linux-gnu-5.2.4";
   LUA_BINDIR = "/nix/store/29p4fvkmbr6hy91304dcsragir7k7bmm-lua-aarch64-unknown-linux-gnu-5.2.4/bin";
   LUA_VERSION = "5.2";
   LUA = "/nix/store/29p4fvkmbr6hy91304dcsragir7k7bmm-lua-aarch64-unknown-linux-gnu-5.2.4/bin/lua";
}
```

Reply:
I don't know what the conditions are to have that file populated so maybe it should be `replace-warn`?

Reply:
https://github.com/NixOS/nixpkgs/pull/337752